docs/guide/hello.md

```diff
@@ -0,0 +1,169 @@
+
+## Say "Hello, Starport"
+
+To get your Cosmos SDK blockchain to say "Hello", you will need to modify a protocol buffer file, create a keeper query function that returns data, and register a query function. Protocol buffer files contain proto `rpc`s that define Cosmos SDK queries and message handlers and proto `message`s that define Cosmos SDK types. `rpc`s are also responsible for exposing an HTTP API. [Keeper](https://docs.cosmos.network/v0.42/building-modules/keeper.html) is an abstraction for modifying the state of the blockchain and keeper functions let you query or write to the state. Registering a query needs to happen only once after you add the first query to your chain.
+
+In terms of workflow, developers usually work with proto files first to define Cosmos SDK [messages](https://docs.cosmos.network/v0.42/building-modules/msg-services.html), [queries](https://docs.cosmos.network/v0.42/building-modules/query-services.html), message handlers and then implement the logic of these queries and message handlers in keeper functions.
+
+Create a `posts` query:
+
+```
+starport scaffold query posts --response title,body
+```
+
+`query` accepts a name of the query (in our case, `posts`), an optional list of request parameters (in our case, empty), and an optional comma-separated list of response fields with a `--response` flag (in our case, `body,title`).
+
+The `query` command has created and modified several files:
+
+- modified `proto/hello/query.proto`
+- created `x/hello/keeper/grpc_query_posts.go`
+- modified `x/hello/client/cli/query.go`
+- created `x/hello/client/cli/query_posts.go`
+
+Let's examine some of these changes. For clarity, in the following code blocks we'll skip placeholder comments Starport uses to scaffold code. Don't delete these placeholders, however, to be able to continue using Starport's scaffolding functionality.
+
+In `proto/hello/query.proto` a `Posts` `rpc` has been added to the `Query` `service`.
+
+```
+service Query {
+  rpc Posts(QueryPostsRequest) returns (QueryPostsResponse) {
+    option (google.api.http).get = "/cosmonaut/hello/hello/posts";
+  }
+}
+```
+
+`Posts` `rpc` will be responsible for returning a list of all the posts on chain. It accepts request parameters (`QueryPostsRequest`) and returns response of type `QueryPostsResponse`. `option` defines the endpoint that will be used by gRPC to generate an HTTP API.
+
+Below you can see both request and response types.
+
+```
+message QueryPostsRequest {
+}
+
+message QueryPostsResponse {
+  string title = 1;
+  string body = 2;
+}
+```
+
+`QueryPostsRequest` is empty because requesting all posts doesn't require are parameters. `QueryPostsResponse` contains `title` and `body` that will be returned from the chain.
```

Comment:
```suggestion
`QueryPostsRequest` is empty because requesting all posts doesn't require parameters. `QueryPostsResponse` contains `title` and `body` that will be returned from the chain.
```